lib/arraylist.c

```diff
@@ -54,5 +54,8 @@ void STDCALL sf_array_list_set(ARRAY_LIST *al, void *item, size_t index) {
 }
 
 void* STDCALL sf_array_list_get(ARRAY_LIST *al, size_t index) {
+    if (al->size <= index) {
+        sf_array_list_grow(al, index+1);
+    }
```

Comment:
Same as the previous comment. Without this, uninitialized memory will be read by the following line.